src/server/index.ts

```diff
@@ -0,0 +1,25 @@
+import Fastify from 'fastify';
+import { fastifyAutoload } from '@fastify/autoload';
+import { join } from 'node:path';
+import { config } from '@/config/env';
+import { prettyLog } from '@/utils/prettyLog';
+
+export async function buildServer() {
+  const server = Fastify({
+    logger: {
+      transport: prettyLog,
+    },
+  });
+
+  try {
+    // decorate instance with envVars
+    server.decorate('config', config);
+    server.register(fastifyAutoload, {
+      dir: join(__dirname, '../modules'),
+    });
+  } catch (e: unknown) {
```

Comment:
what you think about use type guards? since you cannot type the error

Reply:
plan to implement this later, in the current scope i can't think of type guard possibilites
will keep this as a reminder.